src/prog_models/prognostics_model.py

```diff
@@ -1085,4 +1087,344 @@ def optimization_fcn(params):
+            Same as in simulate_to_threshold; for DMD, this value is the time step with which the surrogate model is generated  \n
+        data_len: int, optional
+            Value between 0 and 1 that determines fraction of data resulting from simulate_to_threshold that is used to train DMD surrogate model
+            e.g. if data_len = 0.7 and the simulated data spans from t=0 to t=100, the surrogate model is trained on the data from t=0 to t=70 \n        
+        states_dmd: list, optional
+            List of state keys (must match those defined in the PrognosticsModel) to be included in the surrogate model generation \n
+        inputs_dmd: list, optional
+            List of input keys (must match those defined in the PrognosticsModel) to be included in the surrogate model generation \n
+        outputs_dmd: list, optional
+            List of output keys (must match those defined in the PrognosticsModel) to be included in the surrogate model generation \n 
+        events_dmd: list, optional
+            List of event state keys (must match those defined in the PrognosticsModel) to be included in the surrogate model generation \n           
+        
+        Returns
+        -------
+        SurrogateModel(): class
+            Instance of SurrogateModel class
+
+        Example
+        -------
+        See examples/generate_surrogate
+        
+        """
+
+        # Generate Data to train surrogate model: 
+
+        # Configure
+        config = { # Defaults
+            't0': 0.0,
+            'dt': 1.0, 
+            'save_freq': 1.0, 
+            'horizon': 1e100, # Default horizon (in s), essentially inf
+            'print': False,
+            'progress': False,
+            'data_len': 1,
+            'states_dmd': self.states,
+            'inputs_dmd': self.inputs,
+            'outputs_dmd': self.outputs,
+            'events_dmd': self.events
+        }
+        config.update(kwargs)
+
+        # List of user-define values to include in surrogate model: 
+        states_dmd = deepcopy(self.states)
+        inputs_dmd = deepcopy(self.inputs)
+        outputs_dmd = deepcopy(self.outputs)
+        events_dmd = deepcopy(self.events)
+
+        # Check if user set save_pts, exception
+        if 'save_pts' in config.keys():
+            raise ProgModelInputException("'save_pts' is not a valid input for DMD Surrogate Model.")
+
+        # Initialize lists to hold individual matrices
+        x_list = []
+        xprime_list = []
+        time_list = []
+
+        for iter_load in range(len(load_functions)):
+            # Print status
+            print('Generating training data: loading profile {} of {}'.format(iter_load+1, len(load_functions)))
+
+            # Define current loading function 
+            load_fcn_now = load_functions[iter_load]
+
+            # Simulate to threshold 
+            (times, inputs, states, outputs, event_states) = self.simulate_to_threshold(load_fcn_now, **config)
```

Comment:
Maybe use namedtuple syntax for `simulate_to_threshold` return? Not sure if this was implemented in the namedtuple PR @teubert

Reply:
I think it's fine either way. 

We changed the examples to the named tuple syntax to clean them up/encourage that use. But here internal to a feature it's fine either way

Reply:
For now, I'll leave as-is (due to my lack of python knowledge - @lawrence-hwang maybe you can explain this to me in the future!)